core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala

```diff
@@ -194,7 +194,9 @@ class ResourceProfile(
       throw new SparkException("No executor resource configs were not specified for the " +
         s"following task configs: ${taskResourcesToCheck.keys.mkString(",")}")
     }
-    logInfo(s"Limiting resource is $limitingResource at $taskLimit tasks per executor")
+    if (limitingResource.nonEmpty) {
+      logInfo(s"Limiting resource is $limitingResource at $taskLimit tasks per executor")
```

Comment:
This result in *20/02/25 21:47:55 INFO ResourceProfile: Limiting resource is  at -1 tasks per executor* when `shouldCheckExecCores` is `false`.

Reply:
Again I'm ok with removing it but don't see the problem with it as is either, though could be easier to read when empty, it tells you something about the code and what it did which is the point.

Reply:
I'm not going to remove it but only avoid it when we don't check executor cores. Because when `shouldCheckExecCores` is `false`, `limitingResource` is empty and `taskLimit` is -1.
Do you really intentionally to log this?

BTW, I don't think end-user would learn anything related to source code from the log and they may even don't care about.

Reply:
yes I meant to leave it as it tells you something.  Yes the way it prints could definitely be more readable in the empty case.   The lack of it being there can tell you the same thing but to me it's much easier to search the logs for something that is always there. 

Depends on who your end user is. If you are support person helping end user then I see it as possibly useful which is why I made it an info log.  I figure with this being new and perhaps people don't configuring these properly users can use this to see why they might not be getting as many tasks per executor as they want.  So from that point they would care about it.